app/src/sm_uart_handler.c

```diff
@@ -49,12 +52,30 @@ K_MSGQ_DEFINE(rx_event_queue, sizeof(struct rx_event_t), UART_RX_EVENT_COUNT, 4)
 RING_BUF_DECLARE(tx_buf, CONFIG_SM_UART_TX_BUF_SIZE);
 K_MUTEX_DEFINE(mutex_tx_put); /* Protects the tx_buf from multiple writes. */
 
-enum uart_recovery_state {
-	RECOVERY_IDLE,
-	RECOVERY_ONGOING,
-	RECOVERY_DISABLED
+enum sm_uart_state {
+	SM_TX_ENABLED_BIT,
+	SM_RX_ENABLED_BIT,
+	SM_RX_RECOVERY_BIT,
+	SM_RX_RECOVERY_DISABLED_BIT
+};
```

Comment:
I'd add the enum name into variables so it's easier to read the code when not knowing all the details.
So I mean `SM_TX_ENABLED_BIT` -> `SM_UART_STATE_TX_ENABLED_BIT`.

Similarly for `sm_pipe_state`.